lighty-core/lighty-codecs/src/main/java/io/lighty/codecs/DataCodec.java

```diff
@@ -64,30 +69,33 @@
         XML_FACTORY.setProperty(XMLOutputFactory.IS_REPAIRING_NAMESPACES, true);
     }
 
-    private final BindingToNormalizedNodeCodec codec;
+    private final AdapterContext codec;
     private final DeserializeIdentifierCodec deserializeIdentifierCodec;
     private final SerializeIdentifierCodec serializeIdentifierCodec;
     private final JsonNodeConverter jsonNodeConverter;
     private final XmlNodeConverter xmlNodeConverter;
-    private final SchemaContext schemaContext;
+    private final EffectiveModelContext effectiveModelContext;
 
-    public DataCodec(final SchemaContext schemaContext) {
-        this.schemaContext = requireNonNull(schemaContext);
-        final BindingNormalizedNodeCodecRegistry registry = new BindingNormalizedNodeCodecRegistry();
-        this.codec = new BindingToNormalizedNodeCodec(GeneratedClassLoadingStrategy.getTCCLClassLoadingStrategy(),
-            registry);
-        this.codec.onGlobalContextUpdated(schemaContext);
+    public DataCodec(final EffectiveModelContext effectiveModelContext) {
+        this.effectiveModelContext = requireNonNull(effectiveModelContext);
+        DefaultBindingRuntimeGenerator bindingRuntimeGenerator = new DefaultBindingRuntimeGenerator();
+        BindingRuntimeTypes runtimeTypes = bindingRuntimeGenerator.generateTypeMapping(effectiveModelContext);
+        BindingRuntimeContext bindingRuntimeContext = DefaultBindingRuntimeContext.create(runtimeTypes,
+                GeneratedClassLoadingStrategy.getTCCLClassLoadingStrategy());
+        BindingCodecContext bindingCodecContext = new BindingCodecContext(bindingRuntimeContext);
 
-        this.deserializeIdentifierCodec = new DeserializeIdentifierCodec(schemaContext);
-        this.serializeIdentifierCodec = new SerializeIdentifierCodec(schemaContext);
-        this.xmlNodeConverter = new XmlNodeConverter(this.schemaContext);
-        this.jsonNodeConverter = new JsonNodeConverter(this.schemaContext);
+        this.codec = new ConstantAdapterContext(bindingCodecContext);
```

Comment:
since we have a constant context, we probably want to do a .currentSerializer() here